diff --git a/vendor/github.com/containerd/containerd/v2/internal/cri/server/container_checkpoint_linux.go b/vendor/github.com/containerd/containerd/v2/internal/cri/server/container_checkpoint_linux.go
index 0e207d6..34c1346 100644
--- a/vendor/github.com/containerd/containerd/v2/internal/cri/server/container_checkpoint_linux.go
+++ b/vendor/github.com/containerd/containerd/v2/internal/cri/server/container_checkpoint_linux.go
@@ -71,6 +71,9 @@ func (c *criService) checkIfCheckpointOCIImage(ctx context.Context, input string
 
 	image, err := c.LocalResolve(input)
 	if err != nil {
+		if errdefs.IsNotFound(err) {
+			return "", nil
+		}
 		return "", fmt.Errorf("failed to resolve image %q: %w", input, err)
 	}
 	containerdImage, err := c.toContainerdImage(ctx, image)
